allure-commandline/src/main/java/ru/yandex/qatools/allure/utils/DeleteVisitor.java

```diff
@@ -0,0 +1,33 @@
+package ru.yandex.qatools.allure.utils;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.nio.file.FileVisitResult;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.SimpleFileVisitor;
+import java.nio.file.attribute.BasicFileAttributes;
+
+/**
+ * The visitor deletes files and directories.
+ */
+public class DeleteVisitor extends SimpleFileVisitor<Path> {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DeleteVisitor.class);
+
+    @Override
+    public FileVisitResult visitFile(Path file, BasicFileAttributes attrs) throws IOException {
+        Files.delete(file);
+        LOGGER.debug("Delete <{}>", file);
+        return FileVisitResult.CONTINUE;
+    }
+
+    @Override
+    public FileVisitResult postVisitDirectory(Path dir, IOException exc) throws IOException {
+        Files.delete(dir);
```

Comment:
> Or it throws exception otherwise

It's ok. The `postVisitDirectory` called after all child files are visited. So the directory always should be an empty, or we should throw an exception